packages/stream-api/src/webSocketClient.ts

```diff
@@ -89,7 +91,9 @@ export class WebSocketClient {
     this.reopenAttempt += 1;
 
     try {
-      await this.connect(urlProvider);
+      if (!this.retryDisabled) {
+        await this.connect(urlProvider);
+      }
```

Comment:
A better way to do this would be to use `clearTimeout` for the timer that's created above.

Reply:
Not sure which timer you are referring to here.

Reply:
It's the timer id that's returned on line 83. Instead you should store the ID of that timer, and then call `clearTimeout` when `close` is called:

```js
const timer = setTimeout(() => console.log("hi"), 100);
clearTimeout(timer);
```

Reply:
I have updated this to save the timer, and clear it on a close event.